spec/controllers/labels_controller_spec.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe LabelsController do
+  render_views
+
+  let(:user) { create(:user) }
+  let(:label) { create(:label) }
+
+  describe 'GET #search' do
+    context 'without being signed in' do
+      it 'redirects to other page' do
+        get :search, params: {search: label.name, page: 1}
+        expect(response.body).to match(%r{^<html><body>You are being.*redirected.*</body></html>$})
```

Comment:
Rather than checking for the HTTP response no one should ever read, I would check for the actual redirect. What do you think of these expectations:

```rb
expect(response).to have_http_status(:redirect)
expect(response).to redirect_to(root_url)
```

Reply:
I was not aware of the redirect_to() expectation but that is much cleaner of course.